File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    answer = ''
    st = []
    cnt = 0
    for n in number:
        if cnt == k:
            st.append(n)
            continue
        while st and st[-1] < n:
            st.pop()
            cnt += 1
            if cnt == k:
                break
        st.append(n)
    if cnt < k:
        st = st[:-(k - cnt)]
    return "".join(st)

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.swift
import Foundation

func solution(_ number:String, _ k:Int) -> String {
    var st = [Character]()
    var k = k
    
    for num in number {
        while k > 0 && !st.isEmpty && st.last! < num {
            st.removeLast()
            k -= 1
        }
        st.append(num)
    }
    
    if k > 0 {
        st.removeLast(k)
    }
    
    return String(st)
}